controllers/reservation.controllers.js

```diff
@@ -83,47 +81,473 @@ const listReservationByDate = async ({ client, ack, say, body }) => {
       },
       blocks: modalContent,
     },
   });
 };
 
-const newModal = () => {
-  return [
-    {
-      type: "header",
-      text: {
-        type: "plain_text",
-        text: "",
-        emoji: true,
-      },
-    },
-    {
-      type: "divider",
+const newModal = () => [
+  {
+    type: 'header',
+    text: {
+      type: 'plain_text',
+      text: '',
+      emoji: true,
     },
-  ];
+  },
+  {
+    type: 'divider',
+  },
+];
+
+const newUserCard = (name, email, profilePhoto) => [
+  {
+    type: 'image',
+    image_url: profilePhoto,
+    alt_text: email,
+  },
+  {
+    type: 'plain_text',
+    text: name,
+    emoji: true,
+  },
+];
+
+const addReservation = async ({
+  client, ack, say, body,
+}) => {
+  const options = await getRooms(body.user_id, say);
+  await ack();
+
+  const { text } = body;
+
+  try {
+    // Call views.open with the built-in client
+    if (text) {
+      const command = text.split(' ');
+      const date = command[0];
+      const roomName = command[1];
+      const slackId = body.user_id;
+      const user = await User.findOne({ slackId });
+      const currentDate = moment();
+
+      if (!user) {
+        say(':scream: - Error: User not found.');
+        return;
+      }
+
+      if (!moment(date, 'DD/MM/YYYY').isValid()) {
+        say(
+          ':upside_down_face: *- Hey there, date format must be DD/MM/YYYY!*',
+        );
+        return;
+      }
+
+      const formatedDate = new moment(date, 'DD/MM/YYYY').toDate();
+      const invalidDate = !date || moment(formatedDate).diff(currentDate, 'days') < 0;
+
+      if (invalidDate) {
+        say(':upside_down_face: *- La fecha seleccionada no es válida!*');
+        return;
+      }
+
+      const room = await checkRoomExistence(body.user_id, say, roomName);
+      if (!room) {
+        say(':upside_down_face: *- No se encontró una sala con ese nombre');
+        return;
+      }
+
+      const startOfWeek = moment(formatedDate).startOf('isoWeek');
+      const endOfWeek = moment(formatedDate).endOf('isoWeek');
+      const query = {
+        date: { $gt: startOfWeek, $lt: endOfWeek },
+        user: user._id,
+      };
+      const currentOffice = await Office.findById(user.office);
+
+      const weekResevationByUser = await Reservation.find(query);
+
+      const alreadyHaveReserve = await Reservation.find({
+        date: formatedDate,
+        office: user.office,
+        room: room._id,
+        user: user._id,
+      });
+
+      const currentReservations = await Reservation.find({
+        date: formatedDate,
+        office: user.office,
+        room: room._id,
+      });
+
+      const isRoomFull = currentReservations.length
+        && currentReservations.length >= currentReservations[0].room.maxCapacity;
+
+      if (isRoomFull) {
+        say(
+          ':upside_down_face: *- La sala seleccionada no tiene más horarios disponibles',
+        );
+        return;
+      }
+
+      if (alreadyHaveReserve.length) {
```

Comment:
any reason for this to be here and not immediately after declaring the variable?